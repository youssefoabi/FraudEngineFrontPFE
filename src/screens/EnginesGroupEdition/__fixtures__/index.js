export default {
  groupId: 1,
  groupCode: 2,
  groupName: 'groupName',
  isEnable: true,
  engines: [
    {
      engineId: 1,
      engineName: 'engine 1',
      order: 1,
      engineLabel: 'engineLabel',
      pictogramUrl: 'pictogramUrl',
      isEnable: true,
    },
    {
      engineId: 2,
      engineName: 'engine 2',
      order: 0,
      engineLabel: 'engineLabel 2',
      pictogramUrl: 'pictogramUrl 2',
      isEnable: false,
    },
  ],
  createdAt: '2020-06-16T15:35:00.342Z',
  modifiedAt: '2020-06-16T15:35:00.342Z',
};
