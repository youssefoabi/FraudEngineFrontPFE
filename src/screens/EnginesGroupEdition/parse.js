import { fromPairs, isNil, map, pipe, prop } from 'ramda';
import { isNilOrEmpty } from 'ramda-adjunct';

const toTimestamp = date => (date ? Number(new Date(date)) : null);

const parseEnginesGroup = enginesGroup => {
  if (isNil(enginesGroup)) return null;

  const { groupId, groupCode, groupName, isEnable, engines, createdAt, modifiedAt } = enginesGroup;

  const mapEngines = allEngines => {
    if (isNilOrEmpty(allEngines)) return null;

    return pipe(
      map(engine => [prop('engineId', engine), engine]),
      fromPairs,
    )(allEngines);
  };

  return {
    groupId,
    groupCode,
    groupName,
    isEnable,
    engines: mapEngines(engines),
    createdAt,
    modifiedAt: toTimestamp(modifiedAt),
  };
};
export default parseEnginesGroup;
