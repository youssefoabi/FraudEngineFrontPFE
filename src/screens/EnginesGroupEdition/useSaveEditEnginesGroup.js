import { useCallback } from 'react';
import { useDispatch } from 'react-redux';

import { EDIT_ENGINESGROUP_REQUEST } from '../../components/EnginesGroupForm/actions';

export default function useSaveEditEnginesGroup() {
  const dispatch = useDispatch();

  const saveEditEnginesGroup = useCallback(
    enginesGroup => dispatch({ type: EDIT_ENGINESGROUP_REQUEST, payload: { enginesGroup } }),
    [dispatch],
  );

  return { saveEditEnginesGroup };
}
