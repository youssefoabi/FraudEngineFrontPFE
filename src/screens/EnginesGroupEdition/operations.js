import { call, put, takeLatest } from 'redux-saga/effects';

import {
  EDIT_ENGINESGROUP_ERROR,
  EDIT_ENGINESGROUP_REQUEST,
  EDIT_ENGINESGROUP_SUCCESS,
} from '../../components/EnginesGroupForm/actions';
import { PENDING_REQUEST, RESET_PENDING_REQUEST } from '../../components/ServerActionBackdrop/actions';
import clientApi from '../../client-api';
import parse from '../EnginesGroupCreation/parse';

export function* saveEnginesGroup({ payload }) {
  yield put({ type: PENDING_REQUEST });

  try {
    const { enginesGroup } = payload;
    if (!enginesGroup) return null;
    yield call(clientApi().updateEnginesGroup, parse(enginesGroup));

    yield put({ type: RESET_PENDING_REQUEST });
    yield put({ type: EDIT_ENGINESGROUP_SUCCESS });
  } catch (error) {
    yield put({ type: RESET_PENDING_REQUEST });
    yield put({
      type: EDIT_ENGINESGROUP_ERROR,
      error: { errorMessage: "Erreur d'enregistrement du groupe de moteurs" },
    });
  }

  return true;
}

export default [takeLatest(EDIT_ENGINESGROUP_REQUEST, saveEnginesGroup)];
