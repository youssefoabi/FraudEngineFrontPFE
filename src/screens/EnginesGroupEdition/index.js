import { pathOrNull } from '@cdiscount/ramda-utils';
import { prop } from 'ramda';
import React, { useEffect, useState } from 'react';
import useReactRouter from 'use-react-router';

import clientApi from '../../client-api';
import EnginesGroupScreen from '../EnginesGroupScreen';
import parseEnginesGroup from './parse';
import useSaveEditEnginesGroup from './useSaveEditEnginesGroup';

function EnginesGroupEdition() {
  const [enginesGroup, setEnginesGroup] = useState({});
  const { match } = useReactRouter();

  const { saveEditEnginesGroup } = useSaveEditEnginesGroup();

  useEffect(() => {
    const id = pathOrNull(['params', 'id'], match);
    clientApi()
      .fetchEnginesGroup(id)
      .then(prop('data'))
      .then(rawEnginesGroup => {
        setEnginesGroup(parseEnginesGroup(rawEnginesGroup));
      });
  }, [match]);

  return (
    <EnginesGroupScreen
      enginesGroup={enginesGroup}
      action={saveEditEnginesGroup}
      title="Ã©dition"
      setEnginesGroup={setEnginesGroup}
    />
  );
}

export default EnginesGroupEdition;
