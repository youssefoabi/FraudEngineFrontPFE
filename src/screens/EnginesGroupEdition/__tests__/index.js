import { itRendersAllMutations } from '@cdiscount/jest-utils';

import clientApi from '../../../client-api';
import EnginesGroupEdition from '../';

jest.mock('../../../client-api');

beforeEach(() => {
  clientApi.mockClear();
});

jest.mock('react-redux', () => ({
  useDispatch: () => ({}),
}));

jest.mock('use-react-router', () => () => ({ match: {} }));

jest.mock('react-beautiful-dnd', () => ({
  Droppable: ({ children }) =>
    children(
      {
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      },
      {},
    ),
  Draggable: ({ children }) =>
    children(
      {
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      },
      {},
    ),
  DragDropContext: ({ children }) => children,
}));

describe(EnginesGroupEdition.name, () => {
  itRendersAllMutations(EnginesGroupEdition, [{ name: 'without props', props: {} }]);
});
