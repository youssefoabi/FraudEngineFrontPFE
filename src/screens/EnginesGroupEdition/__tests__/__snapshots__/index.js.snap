// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnginesGroupEdition EnginesGroupEdition renders correctly without props 1`] = `
<EnginesGroupScreen
  action={[Function]}
  enginesGroup={Object {}}
  setEnginesGroup={[Function]}
  title="Ã©dition"
/>
`;
